src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -831,10 +854,14 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
+            });
+            // TODO: this should be returning S_ASYNC_STOP
```

Comment:
hmm... I am a little concerned that we aren't going to get to this for update 3. Well, hopefully this isn't too much of a problem.